.github/workflows/build-docker-image.yml

```diff
@@ -0,0 +1,106 @@
+# To test: gh workflow run 'Build Docker Image' --ref kwannoel/workflow-update-branch
+name: Build Docker Image
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag for the Docker image: vX.Y.Z-<tag>-<commit_sha>'
+        required: true
+        type: string
+        default: 'untagged'
+      # By default, images should be built for the latest commit of a release-branch.
+      # This is disabled by default.
+      # Only in special circumstances when we need an urgent patch,
+      # before some commit is merged into a release branch,
+      # we can set the `enable_patch_build` input to `true`.
+      enable_patch_build:
+        description: 'Allow patch builds on release branches, default: false'
+        required: true
+        type: boolean
+        default: false
+      # By default, images should be built for the latest commit of a release-branch.
+      # This should only be used to build test images,
+      # and never for production.
+      # Images will be tagged with <commit_sha>-<tag>, to enforce that the image is not used in production.
+      # Since the `commit_sha` will never match semver (vX.Y.Z).
+      enable_alpha_build:
+        description: 'Allow custom build for any branch, default: false'
+        required: true
+        type: boolean
+        default: false
+
+jobs:
+  # Generate image tag using:
+  # - version inside Cargo.toml,
+  # - the commit SHA,
+  # - and the tag provided by the user.
+  generate_image_tag:
+    runs-on: ubuntu-latest
+    # Checks that the branch is a release branch, by checking that the semver
+    # should match: vX.Y.Z
+    steps:
+      - uses: actions/checkout@v4
+      - name: 'Print inputs'
+        run: |
+          echo "TAG=${{ github.event.inputs.tag }}"
+          echo "enable_patch_build=${{ github.event.inputs.enable_patch_build }}"
+          echo "enable_alpha_build=${{ github.event.inputs.enable_alpha_build }}"
+      - name: 'Verify release branch'
+        id: verify_release_branch
+        if: github.event.inputs.enable_patch_build == 'false' && github.event.inputs.enable_alpha_build == 'false'
+        run: |
+          # Get current branch name
+          echo "GITHUB_REF=$GITHUB_REF"
+          BRANCH_NAME=$(basename $GITHUB_REF)
```

Comment:
What if my branch is named `bz/release-2.4` 🤔 Is there any better way to get the branch name?

Reply:
Fixed, used `git rev-parse` instead.